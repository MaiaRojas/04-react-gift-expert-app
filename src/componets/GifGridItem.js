import React from 'react';
import PropTypes from 'prop-types'


export const GifGridItem = ({ title, url}) => { 
  return (
    <div className="card animate__fadeIn">
      <p>{ title }</p>
      <img src={url} alt={title} />
    </div>
  )
};

GifGridItem.protoTypes = {
  title: PropTypes.string.isRequired,

}