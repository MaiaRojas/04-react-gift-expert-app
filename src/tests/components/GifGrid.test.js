import React from 'react'
import { shallow } from "enzyme";
import { GifGrid } from "../../componets/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
//import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')


describe('Test in <GifGrid />', () => {
  const category ='Rick and Morty';

  test('should return a snapshots', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should display items when images page are uploaded', () => {
    const gifs = [{
      id: 'abc',
      url: 'https://localhost/anything/thing.jpg',
      title: 'Anything'
    }];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });


  test('should display items when images(more images) page are uploaded', () => {
    const gifs = [{
      id: 'foo',
      url: 'https://localhost/anything/thing.jpg',
      title: 'Anything'
    },
    {
      id: 'bar',
      url: 'https://localhost/anything/bar.jpg',
      title: 'Bar'
    }
  ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });

})