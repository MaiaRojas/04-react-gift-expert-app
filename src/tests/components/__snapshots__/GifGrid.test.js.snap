// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGrid /> should display items when images page are uploaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rick and Morty
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Anything"
      url="https://localhost/anything/thing.jpg"
    />
  </div>
</Fragment>
`;

exports[`Test in <GifGrid /> should display items when images(more images) page are uploaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rick and Morty
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="foo"
      key="foo"
      title="Anything"
      url="https://localhost/anything/thing.jpg"
    />
    <GifGridItem
      id="bar"
      key="bar"
      title="Bar"
      url="https://localhost/anything/bar.jpg"
    />
  </div>
</Fragment>
`;

exports[`Test in <GifGrid /> should return a snapshots 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Rick and Morty
  </h3>
  <p
    className="animate__animated animate__fadeIn"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
